Mogakco/Sources/Domain/UseCases/CreateStudyUseCase.swift

```diff
@@ -22,11 +22,21 @@ struct CreateStudyUseCase: CreateStudyUseCaseProtocol {
         self.userRepository = userRepository
     }
     
-    func create(study: Study) -> Observable<Study> {
+    func create(study: Study) -> Observable<Void> {
         var study = study
         return userRepository.load()
             .compactMap { $0.id }
             .map { study.userIDs.append($0) }
```

Comment:
```suggestion
        let study = study
        return userRepository.load()
            .compactMap { $0.id }
            .map { Study(study: study, userIDs: userIDs) }
```
Study 프로퍼티를 variable로 변경하는 것과 새로운 스터디를 생성해서 반환하는 방법 중 어떤 방식이 좋을지 고민되네요
side effect까지 발생하는 부분으로 보이지는 않지만 함수형 프로그래밍에서는 variable 사용을 지양하기 때문에 entity에 variable변수가 사용되는게 조금 걱정되는 부분인 것 같아요!

Reply:
맞아요. 함수형 프로그래밍으로 코드를 짰으면 한눈에 보고 이해할 수 있어야 하는데 variable을 사용하니까 이해하기 어렵더라구요. 이 부분은 [ChatRoom, Study의 userIDs 업데이트 API 구현](https://github.com/boostcampwm-2022/iOS04-Mogakco/pull/244#issue-1464829240) PR이 머지되면 API를 이용해서 Study의 userIDs 필드를 업데이트하거나 아니면 Study에 이니셜라이저를 추가하는 방향으로 리팩토링하겠습니다!